### ISSUE TITLE
Remove window title bar, but save frame e.t.c

### ISSUE THREAD

**Issue by NotASocol** [glfw/glfw#1518]
Is it possible to remove title bar (with title and control buttons like close e.t.c) on window, but save window frame and other effects?

Undecorated window has nothing, not only title bar.

I need something like that (on Windows), not just floating panel.
https://imgur.com/a/x6ZSLrK

**Comment by ghuser404** [comment]
You can use undecorated window and implement your own decorations in the client area.

**Comment by NotASocol** [comment]
> You can use undecorated window and implement your own decorations in the client area.

Yes, but it is pretty hard to add some decorations like window shadow. It is need to be outside the window (not at render contex)

**Comment by ghuser404** [comment]
True. But I don't think you can draw custom widgets in the title bar, either way.
For shadows you could have a small margin within your render context.

**Comment by etisdew** [comment]
There is a route for removing the titlebar and leaving a frame but I can't put my finger on when I did it last or how and just relying on keyboard commands to close and minimize. I don't know if thats even an option for you either. Faking it seems to be a preferable methodology expecially considering multi-targetability, for clarity what is the convention for creating an undecorated window? Perhaps there is a windows specific solution to meet your requirements but I doubt it would be any less effort than some clever positioning.

**Comment by AZProductions** [comment]
Any updates on this?

**Comment by Speiger** [comment]
@elmindreda @AZProductions A fork has implemented custom titlebar support.
https://github.com/TheCherno/glfw
With a video showcasing it.
https://www.youtube.com/watch?v=-NJDxf4XwlQ

Maybe that could be merged into the official branch?
